projects/bms_carrier/src/current_sense.c

```diff
@@ -1,6 +1,80 @@
 #include "current_sense.h"
 
-// TODO(SOFT-172): Actually implement this method
+#include "ads1259_adc.h"
+#include "bms.h"
+#include "exported_enums.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static CurrentReadings *s_readings;
```

Comment:
I think it's worth having a no-argument `current_sense_is_charging`, but switched to just save a static bool rather than hold onto the main pointer. Also switched readings to a `CurrentStorage` with associated settings included.